src/store.rs

```diff
@@ -150,46 +232,23 @@ impl Store {
             })?;
         }
 
-        // Finish up by making sure everything is readonly and store it in the
-        // final location.
-        for dir in &output_dirs {
+        // Now that we're all done moving files over and making them read-only,
+        // we can safely make all the directories read-only too.
+        for dir in &created_dirs {
             Self::make_readonly(&temp.path().join(&dir)).with_context(|| {
                 format!("could not make `{}` read-only in the store", dir.display(),)
             })?;
         }
 
-        let final_hash = hasher.finalize().to_hex().to_string();
-        let final_location = self.root.join(&final_hash);
-
-        if final_location.exists() {
-            log::debug!(
-                "we have already stored {}, so I'm skipping the move!",
-                final_hash
-            );
-        } else {
-            log::debug!("moving {} into store", &final_hash);
-
-            // important: at this point we need to take ownership of the tempdir so
-            // that it doesn't get automatically removed when it's dropped. We've
-            // so far avoided that to avoid leaving temporary directories laying
-            // around in case of errors.
-            std::fs::rename(temp.into_path(), &final_location)
-                .context("could not move temporary collection dir into the store")?;
-            Self::make_readonly(&final_location).context("could not make store path readonly")?;
-        }
+        // important: at this point we need to take ownership of the tempdir so
+        // that it doesn't get automatically removed when it's dropped. We've
+        // so far avoided that to avoid leaving temporary directories laying
```

Comment:
I guess this depends on whether on not the tmp dir being left around would be useful for debugging.

Reply:
I could see the workspace dir being useful for debugging a failed job, and there's some easy extension points in `Workspace` to allow that to happen. This temp dir, though, wouldn't be very interesting—it's just for moving stuff around.

I might be wrong, but I'd kind of prefer to wait and see how it shakes out when using the system, since we'll need to address error cases as a whole at some point. Does that sound reasonable to you?

Reply:
I'm going to move forward with removing this all the time for now. We can change it later if we need to.

Reply:
Oh never mind, I can't. Need an approval. 😆

Reply:
Approval given!

Reply:
Thank you!